src/modules/qt/gps_parser.cpp

```diff
@@ -767,156 +851,159 @@ void xml_parse_tcx(xmlNodeSetPtr found_nodes, gps_point_ll **gps_list, int *coun
 		</HeartRateBpm>
 	  </Trackpoint>
 	*/
-	//mlt_log_info(NULL, "xml_parse_tcx - parsing %d elements\n", found_nodes->nodeNr);
-
-	for (i=0; i<found_nodes->nodeNr; i++)
+	while (!reader.atEnd() && !reader.hasError())
 	{
-		gps_point_raw crt_point = uninit_gps_raw_point;
-		crt_node = found_nodes->nodeTab[i];
-
-		for (gps_val = crt_node->children; gps_val; gps_val = gps_val->next)
+		reader.readNext();
+		// mlt_log_info(NULL, " readNextStartElement tag: %s \n", _qxml_getthestring(reader.name()));
+		if (reader.isStartElement() && reader.name() == "Trackpoint")
```

Comment:
Qt6 `../src/modules/qt/gps_parser.cpp:858:51: error: invalid conversion from ‘const char*’ to ‘char16_t’ [-fpermissive]`